app/models/reaction.rb

```diff
@@ -66,7 +66,14 @@ def cached_any_reactions_for?(reactable, user, category)
   # - receiver is the same user as the one who reacted
   # - receive_notification is disabled
   def skip_notification_for?(_receiver)
-    points.negative? || (user_id == reactable.user_id)
+    reactor_id = case reactable
+                 when User
+                   reactable.id
+                 else
+                   reactable.user_id
+                 end
```

Comment:
This is really neat and very readable! Interesting that you don't need to do something like this:
```ruby
case reactable.class
when User
```

I wonder how the logic knows that `reactable == User` 🤔

Reply:
ActiveRecord is sneaky like that

```ruby
      # Overwrite the default class equality method to provide support for decorated models.
      def ===(object) # :nodoc:
        object.is_a?(self)
      end
```

Reply:
Ahh bingo. Nice digging 🕵️

Reply:
Unlike similar constructs in other languages, the `case` keyword in Ruby uses an operator that implies a kind of fuzzy match — it's `===` in Ruby. It is _amaaaazing_. Mechanically, the value after each `when` clause gets `===` called with the value after the `case` (`when_expression === case_expression`), so in this PR, it's basically equivalent to this:

```ruby
if User === reactable
  reactable.id
else
  reactable.user_id
end
```

It lets you do things like this:

```ruby
case my_string
when /something i want to match on/
  puts "matched with a regex"
when "omg"
  puts "matched by value equality"
when proc { |value| value.start_with? "hello" }
  puts "matched with a proc"
when "a"..."z"
  puts "matched with a range"
else
  puts "no match"
end
```

You can pass any object to a `when` clause that has an `===` method and it'll work.

```ruby
class HasUserID
  def initialize(value)
    @value = value
  end

  def ===(other)
    if other.respond_to? :user_id
      @value == other.user_id
    else
      @value == other.id
    end
  end
end

require "ostruct"
user = OpenStruct.new(id: 123)
article = OpenStruct.new(user_id: 123)
article_from_another_author = OpenStruct.new(user_id: 1234)

[user, article, article_from_another_author].each do |obj|
  case obj
  when HasUserID.new(123)
    pp obj => "yep"
  else
    pp obj => "nope"
  end
end
# {#<OpenStruct id=123>=>"yep"}
# {#<OpenStruct user_id=123>=>"yep"}
# {#<OpenStruct user_id=1234>=>"nope"}
```

Reply:
Amazing, TIL! Sounds like it's much better than an `if` or a ternary statement, and arguably more readable in some cases (no pun intended).